mantidproject/mantid
# Allow use of True and False in Option Text Entry in the Refl GUI

Requested by user: Cristy and Andrew on POLREF

**Description of work.**
Since the Refl GUI is in C++, it can make calls to the C++ algorithms
fairly directly. This means that when we give it string values like
True and False they don't seem to get converted to bools properly. This
creates confusing behaviour when the user wants to be able to write in
their properties like they would in the python scripts. Instead, we
convert the bool strings to 0 and 1 strings that the algorithm is happy
to process. 

**To test:**
1. Open the Refl GUI
2. Enter Investigation ID: 1120015 Cycle: 11_3 and hit search
3. Select a few adjacent runs and click transfer
4. Go to the Experiment Settings Tab
5. Enter `UseManualScaleFactors=True, ManualScaleFactors=5` into the Output Stitch Properties field
6. Return to the runs tab and select a group with at least two rows.
7. Click process.
8. Find the stitched workspace (Something like: IvsQ_RUNNUM1_RUNNUM2 or IvsQ_RUNNUM1+RUNNUM2_RUNNUM3 or similar)
9. Right click it an click Show History
10. Check that the values you entered were used for Stitch1DMany (Note: True will be converted to 1 and False will be converted to 0)

Fixes #34844 



---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
